mlir/lib/Dialect/PDLInterp/IR/PDLInterp.cpp

```diff
@@ -71,18 +71,20 @@ static ParseResult parseCreateOperationOpAttributes(
   Builder &builder = p.getBuilder();
   SmallVector<Attribute, 4> attrNames;
   if (succeeded(p.parseOptionalLBrace())) {
-    auto parseOperands = [&]() {
```

Comment:
Some comment on parseCreateOperationOpAttributes would help me follow

Reply:
An example for this op is 
`%op0 = pdl_interp.create_operation "foo.op"(%input : !pdl.value) {"attr" = %attribute} -> (%type : !pdl.type)`
The parsing/printing format of this operation is defined [here](https://github.com/Xilinx/llvm-project/blob/63ef598fdcc587113a4229062d41cc275977567f/mlir/include/mlir/Dialect/PDLInterp/IR/PDLInterpOps.td#L453). 
It uses builtin tablegen to parse the operation name and operands + types. Then it refers to custom C++ parsing and printing functions to handle the `OperationOpAttributes`, i.e. `{"attr" = %attribute}` and `OperationOpResults`, i.e. ` -> (%type : !pdl.type)`.
The tablegen `custom<X>(args)` keyword handles dispatch to the corresponding `parseX(args)` and `printX(args)` functions.

The issue that was fixed here is that the braces of `OperationOpAttributes` are omitted if no OperationOpAttributes are present. However, this behavior clashes with the attr-dict that follows the operation in cases where there are also no `OperationOpResults`.

Reply:
Thanks for explaining this! I actually meant adding a comment in the source code to explain this, i.e. saying that we are currently parsing optional attributes like `{"attr" = %attribute}`.

Reply:
added comments 👍